src/k2/module/k23si/Module.cpp

```diff
@@ -870,117 +862,32 @@ K23SIPartitionModule::handleWrite(dto::K23SIWriteRequest&& request, dto::K23SI_M
-            versions.pop_front();
+    if (request.fieldsToUpdate.size() > 0) {
+        // parse the partial record to full record
+        if ( !versions.size() || versions[0].isTombstone) {
+            // cannot parse partial record without a version
+            return RPCResponse(dto::K23SIStatus::KeyNotFound("can not partial update with no/deleted version"), dto::K23SIWriteResponse{});
         }
-        else if (rec.txnId.mtr != rqmtr) {
-            // this is a update request finding a WI from a different transaction. Do another push with the remaining
-            // deadline time.
-            K2DEBUG("Partition: " << _partition << ", different WI found for key " << request.key);
-            sitMTR = rec.txnId.mtr;
-            return _doPush(request.collectionName, rec.txnId, request.mtr, deadline)
-                .then([this, sitMTR, request = std::move(request), deadline](auto&& winnerMTR) mutable {
-                    if (winnerMTR == sitMTR) {
-                        // sitting transaction won. Abort the incoming request
-                        K2DEBUG("Partition: " << _partition << ", push lost for key " << request.key);
-                        return RPCResponse(dto::K23SIStatus::AbortConflict("incumbent txn won in update push"), dto::K23SIPartialUpdateResponse{});
-                    }
-                    // incoming request won. re-run write logic
-                    K2DEBUG("Partition: " << _partition << ", push won for key " << request.key);
-                    return handlePartialUpdate(std::move(request), sitMTR, deadline);
-                });
+        if (!_parsePartialRecord(request, versions[0])) {
+            K2DEBUG("Partition: " << _partition << ", can not parse partial record for key " << request.key);
+            versions[0].value.fieldData.seek(0);
+            return RPCResponse(dto::K23SIStatus::BadParameter("missing fields or can not interpret partialUpdate"), dto::K23SIWriteResponse{});
         }
     }
 
-    // parse the partial record to full record
-    if ( !versions.size() || versions[0].isTombstone) {
-        // cannot parse partial record without a version
-        return RPCResponse(dto::K23SIStatus::KeyNotFound("can not partial update with no/deleted version"), dto::K23SIPartialUpdateResponse{});
-    }
-    if (!_parsePartialRecord(request, versions)) {
-        K2DEBUG("Partition: " << _partition << ", can not parse partial record for key " << request.key);
-        versions[0].value.fieldData.seek(0);
-        return RPCResponse(dto::K23SIStatus::BadParameter("missing fields or can not interpret partialUpdate"), dto::K23SIPartialUpdateResponse{});
-    }
-
     // Clean up if req and WI are in the same transaction
-    if (versions.size() > 0 && versions[0].status == dto::DataRecord::WriteIntent && request.mtr == versions[0].txnId.mtr) {
-        _queueWICleanup(std::move(versions[0]));
+    if (versions.size() > 0 && versions[0].txnId.mtr == request.mtr) {
```

Comment:
I think my answer to the previous question covers this but if not it may be helpful to look at the full function instead of the diff. The diff is hard to read because it is showing the combined handleWrite with some old code from the partial update.